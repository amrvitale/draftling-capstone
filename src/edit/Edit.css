* {
    box-sizing: border-box;
}

.editHeader {
    color: #35322a;
    font-family: 'Playfair Display', 'Georgia', serif;
    text-shadow:
    4px 4px 0 darkorange,
    -1px -1px 0 #000,  
     1px -1px 0 #000,
     -1px 1px 0 #000,
      1px 1px 0 #000;
}
.editHero {
    background-image: url(proofread2.jpg);
    background-position: center; /* Center the image */
    background-repeat: no-repeat; /* Do not repeat the image */
    background-blend-mode: lighten;
}
.postLabel {
    color: whitesmoke;
    
}

.instructions {
    color: #35322a;
    background-color: lightyellow;
    
}