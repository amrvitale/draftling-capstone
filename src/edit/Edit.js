import React from 'react';
import { useParams } from "react-router-dom";
import DraftlingPage from '../draftlingpage/DraftlingPage'
import ApiContext from '../ApiContext'
import {findDraftling} from '../draftlings-helpers'

class Edit extends React.Component {

    componentDidMount() {
        const {draftlings = [] } = this.context
        const {slug} = this.props.match.params
        console.log(slug)
        const editDraftling = draftlings.find(draftling => draftling.id === parseInt(slug))
        console.log(editDraftling)
    }

    static defaultProps = {
        match: {
          params: {}
        }
      }
      static contextType = ApiContext    

    render() {
        this.setState({
            title: editDraftling.title,
           wordcount: editDraftling.wordcount,
           content: editDraftling.content
        })

       

        const { id } = this.props.match.params  
        const styleTextArea = {
            minHeight: '380px',
            minWidth: '380px',
            padding: '9px',
            boxSizing: 'border-box',
            fontSize: '15px',
            textAlign: 'left'
        };

        const styleTitle = {
            minWidth: '200px'
        }
        return (
            <div className="edit">
                 <form onSubmit={this.handleSubmit}>
                    <label htmlFor="draftlingTitle">Draftling Title:</label>
                    <input type="text" name="title" style={styleTitle} onChange={this.handleChange} required></input>
                    <br />
                    <label htmlFor="wordcount">Choose an approximate word count.</label>
                    <select name="wordcount" id="wordcount" onChange={this.handleChange} required>
                        <option></option>
                        <option value="Six word story">Six word story</option>
                        <option value="Minisaga">Minisaga: 50 words</option>
                        <option value="Micro fiction">Microfiction: 51 - 100 words</option>
                        <option value="Twitterature">Twitterature: 280 characters</option>
                        <option value="Sudden Fiction">Sudden Fiction: 300 - 750 words</option>
                        <option value="Flash Fiction">Flash Fiction: 751 - 1000 words</option>
                        <option value="Short Story">Short Story: 1001 - 7500 words</option>
                        <option value="Novellete">Novellete: 7501 - 20,000 words</option>
                        <option value="Novella">Novella: 20,001 - 60,000 words</option>
                        <option value="Novel">Novel: 60,0001 + words</option>
                    </select>
                    <br />
                    <textarea style={styleTextArea} name="content" onChange={this.handleChange} required></textarea>
                    <br/>
                    <input type="submit" value="Submit Draft to My Dash"></input>
                </form>
            </div>
        )
    }
}

export default Edit;