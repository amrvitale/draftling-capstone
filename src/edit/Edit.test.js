import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Edit from './Edit'

describe (`Edit component`, () => {
    it('renders a .Edit by default', () => {
        const wrapper = shallow(<Edit />)
        expect (toJson(wrapper)).toMatchSnapshot()
    })

    
})