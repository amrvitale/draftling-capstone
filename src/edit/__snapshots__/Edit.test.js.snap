// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit component renders a .Edit by default 1`] = `
<div
  className="editPage"
>
  <div
    className="editHero"
  >
    <h1
      className="editHeader"
    >
      Edit Your Draftling
    </h1>
    <p
      className="instructions"
    >
      Edit any fields for your Draftling. 
      <br />
      Upon submitting, it will be placed on your shelf until it's ready to be published.
    </p>
  </div>
  <br />
  <br />
  <form
    onSubmit={[Function]}
  >
    <label
      className="postLabel"
      htmlFor="draftlingTitle"
    >
      Draftling Title:
    </label>
    <input
      name="title"
      onChange={[Function]}
      required={true}
      style={
        Object {
          "minWidth": "200px",
        }
      }
      type="text"
      value=""
    />
    <br />
    <label
      className="postLabel"
      htmlFor="wordcount"
    >
      Choose an approximate word count.
    </label>
    <select
      id="wordcount"
      name="wordcount"
      onChange={[Function]}
      required={true}
      value=""
    >
      <option />
      <option
        value="Six word story"
      >
        Six word story
      </option>
      <option
        value="Minisaga"
      >
        Minisaga: 50 words
      </option>
      <option
        value="Micro fiction"
      >
        Microfiction: 51 - 100 words
      </option>
      <option
        value="Twitterature"
      >
        Twitterature: 280 characters
      </option>
      <option
        value="Sudden Fiction"
      >
        Sudden Fiction: 300 - 750 words
      </option>
      <option
        value="Flash Fiction"
      >
        Flash Fiction: 751 - 1000 words
      </option>
      <option
        value="Short Story"
      >
        Short Story: 1001 - 7500 words
      </option>
      <option
        value="Novellete"
      >
        Novellete: 7501 - 20,000 words
      </option>
      <option
        value="Novella"
      >
        Novella: 20,001 - 60,000 words
      </option>
      <option
        value="Novel"
      >
        Novel: 60,0001 + words
      </option>
    </select>
    <br />
    <label
      className="postLabel"
      htmlFor="genre"
    >
      Choose a genre.
    </label>
    <select
      id="genre"
      name="genre"
      onChange={[Function]}
      required={true}
      value=""
    >
      <option />
      <option
        value="Action/Adventure"
      >
        Action/Adventure
      </option>
      <option
        value="Classic"
      >
        Classic
      </option>
      <option
        value="Contempory"
      >
        Contempory
      </option>
      <option
        value="Crime/Detective"
      >
        Crime/Detective
      </option>
      <option
        value="Fable"
      >
        Fable
      </option>
      <option
        value="Fairy Tale"
      >
        Fairy Tale
      </option>
      <option
        value="Folktale"
      >
        Folktale
      </option>
      <option
        value="Gothic"
      >
        Gothic
      </option>
      <option
        value="Historical Fiction"
      >
        Historical Fiction
      </option>
      <option
        value="Horror"
      >
        Horror
      </option>
      <option
        value="Humor"
      >
        Humor
      </option>
      <option
        value="Magical Realism"
      >
        Magical realism
      </option>
      <option
        value="Mystery"
      >
        Mystery
      </option>
      <option
        value="Romance"
      >
        Romance
      </option>
      <option
        value="Science Fiction"
      >
        Science Fiction
      </option>
      <option
        value="Thriller"
      >
        Thriller
      </option>
      <option
        value="Western"
      >
        Western
      </option>
    </select>
    <br />
    <textarea
      name="content"
      onChange={[Function]}
      required={true}
      style={
        Object {
          "boxSizing": "border-box",
          "fontSize": "15px",
          "minHeight": "380px",
          "minWidth": "380px",
          "padding": "9px",
          "textAlign": "left",
        }
      }
      value=""
    />
    <br />
    <input
      type="submit"
      value="Submit Draft to My Dash"
    />
  </form>
</div>
`;
